XUtils.c

```diff
@@ -364,3 +364,17 @@ double sumPositiveValues(const double* array, size_t count) {
    }
    return sum;
 }
+
+#if !defined(HAVE_BUILTIN_CTZ)
+// map a bit value mod 37 to its position
+static const unsigned int mod37BitPosition[] = {
+  32, 0, 1, 26, 2, 23, 27, 0, 3, 16, 24, 30, 28, 11, 0, 13, 4,
+  7, 17, 0, 25, 22, 31, 15, 29, 10, 12, 6, 0, 21, 14, 9, 5,
+  20, 8, 19, 18
```

Comment:
1. I would like the mapping table to use `uint8_t` type for smaller size.

2. Comment 
```c
// The array can be generated with this shell script:
// {
//    i=0; while [ $i -lt 32 ]; do
//       printf '[%2d] = %2d,\n' $(( (1 << i) % 37)) $i;
//       i=$((i + 1));
//    done;
//    for i in 7 14 19 28; do printf '[%2d] =  0,\n' $i; done
// } | sort
```